internal/datastore/spanner/migrations/driver.go

```diff
@@ -84,3 +88,15 @@ func (smd SpannerMigrationDriver) Close() error {
 }
 
 var _ migrate.Driver = SpannerMigrationDriver{}
+
+func detectEmulatorSetting() (string, bool) {
+	for _, env := range os.Environ() {
```

Comment:
viper knows the env prefix we use - we pass it around as `programName` and configure viper with it:

https://github.com/authzed/spicedb/blob/7bd2af59981f91152009461135b5bb6fd7f304fd/pkg/cmd/server/defaults.go#L53

I'm not overly familiar with viper but it seems like there should be a way to directly ask for this value instead of manually flipping through the environment variables.

Does something like `viper.Get("SPANNER_EMULATOR_HOST")` work? Maybe @jzelinskie has some suggestions?